File: packages/react-native/ReactAndroid/src/main/java/com/facebook/react/views/modal/ShowEvent.java
/*
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

package com.facebook.react.views.modal;

import androidx.annotation.Nullable;
import com.facebook.infer.annotation.Nullsafe;
import com.facebook.react.bridge.Arguments;
import com.facebook.react.bridge.WritableMap;
import com.facebook.react.uimanager.common.ViewUtil;
import com.facebook.react.uimanager.events.Event;

/** {@link Event} for showing a Dialog. */
/* package */ @Nullsafe(Nullsafe.Mode.LOCAL)
class ShowEvent extends Event<ShowEvent> {

  public static final String EVENT_NAME = "topShow";

  @Deprecated
  protected ShowEvent(int viewTag) {
    this(ViewUtil.NO_SURFACE_ID, viewTag);
  }

  protected ShowEvent(int surfaceId, int viewTag) {
    super(surfaceId, viewTag);
  }

  @Override
  public String getEventName() {
    return EVENT_NAME;
  }

  @Nullable
  @Override
  protected WritableMap getEventData() {
    return Arguments.createMap();
  }
}

File: packages/react-native/ReactAndroid/src/main/java/com/facebook/react/views/modal/ShowEvent.kt
/*
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

package com.facebook.react.views.modal

import com.facebook.react.bridge.Arguments
import com.facebook.react.bridge.WritableMap
import com.facebook.react.uimanager.common.ViewUtil
import com.facebook.react.uimanager.events.Event

/** [Event] for showing a Dialog. */
internal class ShowEvent(surfaceId: Int, viewTag: Int) : Event<ShowEvent>(surfaceId, viewTag) {

  @Deprecated(
      "Do not use this constructor, use the one with explicit surfaceId",
      ReplaceWith("ShowEvent(surfaceId, viewTag)"))
  constructor(viewTag: Int) : this(ViewUtil.NO_SURFACE_ID, viewTag)

  override fun getEventName(): String = EVENT_NAME

  override fun getEventData(): WritableMap? = Arguments.createMap()

  companion object {
    const val EVENT_NAME: String = "topShow"
  }
}
